base/linalg/lapack.jl

```diff
@@ -664,7 +675,7 @@ for (gesvx, elty) in
             ferr      = similar(A, $elty, nrhs)
             berr      = similar(A, $elty, nrhs)
             work      = Array($elty, 4n)
-            iwork     = Array($elty, n)
+            iwork     = Array($elty, 2n)
```

Comment:
Integer workspace should be integers, but right now it is a floating point array. Then `BlasInt`s are 64 bit size so the single precision wrapper produces an `iwork` buffer of half the necessary size. Solution `$elty => BlasInt`